### PR TITLE
making docker image and its version configurable

### PR DESCRIPTION
**Pull request by sadovnikov**

### CODE REVIEW

File: logstash-scheduler/src/main/java/org/apache/mesos/logstash/config/LogstashConfig.java [hunk 1]
```diff
@@ -9,9 +9,13 @@
 @Component
 @ConfigurationProperties(prefix = "logstash")
 public class LogstashConfig {
+
     private int heapSize = 64;
     private Optional<URL> elasticsearchUrl = Optional.empty();
 
+    private String executorImage = null;
```
Comment (mwl) [inline logstash-scheduler/src/main/java/org/apache/mesos/logstash/config/LogstashConfig.java]: Shouldn't it default to `mesos/logstash-executor`?

File: logstash-scheduler/src/main/java/org/apache/mesos/logstash/config/LogstashConfig.java [hunk 2]
```diff
@@ -9,9 +9,13 @@
...
+    private String executorVersion = null;
```
Comment (mwl) [inline logstash-scheduler/src/main/java/org/apache/mesos/logstash/config/LogstashConfig.java]: And this default to `latest`?

File: logstash-scheduler/src/main/java/org/apache/mesos/logstash/scheduler/TaskInfoBuilder.java [hunk 3]
```diff
@@ -29,11 +29,16 @@
 
     public Protos.TaskInfo buildTask(Protos.Offer offer) {
 
+        String executorImage = LogstashConstants.EXECUTOR_IMAGE_NAME_WITH_TAG;
+        if (logstashConfig.getExecutorImage() != null && logstashConfig.getExecutorVersion() != null) {
+            executorImage = logstashConfig.getExecutorImage() + ":" + logstashConfig.getExecutorVersion();
+        }
```
Comment (mwl) [inline logstash-scheduler/src/main/java/org/apache/mesos/logstash/scheduler/TaskInfoBuilder.java]: If you add defaults to configuration, you can get rid of this whole bit. Including the entries in `LogstashConstants`

File: logstash-scheduler/src/main/java/org/apache/mesos/logstash/config/LogstashConfig.java [hunk 4]
```diff
@@ -9,9 +9,16 @@
 @Component
 @ConfigurationProperties(prefix = "logstash")
 public class LogstashConfig {
+
+    private static final String DEFAULT_EXECUTOR_IMAGE_NAME = "mesos/logstash-executor";
```
Comment (mwl) [inline logstash-scheduler/src/main/java/org/apache/mesos/logstash/config/LogstashConfig.java]: There is really no need for these private constants. Just inline them on config parameters

Comment (sadovnikov) [inline logstash-scheduler/src/main/java/org/apache/mesos/logstash/config/LogstashConfig.java]: I thought about this, but decided to keep them in order to ease implementation of methods like `isDefaultContainer()`... well, I would keep them if you don't mind. No problems with removing them either. Let me know, plz

Comment (mwl) [inline logstash-scheduler/src/main/java/org/apache/mesos/logstash/config/LogstashConfig.java]: I see what you're thinking but until we have a suge for that feature keeping code at a minimum is our priority. So please inline.

**Comment by mwl** [comment]
Thank you very much!

**Pull request merged**